File: sipXopenfire/config-plugin/plugin/lib/chat_room.dart
library chat_room;

class ChatRoom {
  int id;
  String name;
  String description;
  String subject;
  bool canChangeSubject;
  String password;
  bool public;
  bool moderated;
  bool membersOnly;
  int maxUsers;
  bool canInvite;
  bool canDiscoverJid;
  bool logMessages;
  int rolesToBroadcast;
  List<User> owners;
  List<User> members;

  ChatRoom({
      this.id : -1,
      this.name : "",
      this.description : "",
      this.subject : "",
      this.canChangeSubject : true,
      this.password : "",
      this.public : true,
      this.moderated : false,
      this.membersOnly : false,
      this.maxUsers : 10,
      this.canInvite : true,
      this.canDiscoverJid : false,
      this.logMessages : true,
      this.rolesToBroadcast : 7,
      this.owners : null,
      this.members : null
  });

  factory ChatRoom.fromMap(Map attrs) {
    return new ChatRoom(
        id: attrs['id'],
        name: attrs['name'],
        description: attrs['description'],
        subject: attrs['subject'],
        canChangeSubject: attrs['canChangeSubject'],
        password: attrs['password'],
        public: attrs['public'],
        moderated: attrs['moderated'],
        membersOnly: attrs['membersOnly'],
        maxUsers: attrs['maxUsers'],
        canInvite: attrs['canInvite'],
        canDiscoverJid: attrs['canDiscoverJid'],
        logMessages: attrs['logEnabled'],
        rolesToBroadcast: attrs['rolesToBroadcast'],
        owners: _convert(attrs['owners']),
        members: _convert(attrs['members'])
    );
  }

  static List<User> _convert(List<Map> userMaps) {
    List<User> converted = [];

    if(userMaps != null) {
      for(Map userMap in userMaps) {
        converted.add(new User(userMap));
      }
    }

    return converted;
  }

  get broadcastModerator => rolesToBroadcast & 1 != 0;
  set broadcastModerator(int val) => rolesToBroadcast |= val;

  get broadcastParticipant => rolesToBroadcast & 2 != 0;
  set broadcastParticipant(int val) => rolesToBroadcast |= val;

  get broadcastVisitor => rolesToBroadcast & 4 != 0;
  set broadcastVisitor(int val) => rolesToBroadcast |= val;
}

class User {
  int id;
  String name;
  String displayName;
  int hashCode;

  User(Map attrs) {
    id = attrs['id'];
    name = attrs['name'];
    displayName = attrs['displayName'];
    hashCode = id;
  }

  bool operator ==(User other) => id == other.id;

  String toString() => "[id=" + id.toString() + " name=" + name + " displayName=" + displayName + "]";

  String prettyName() => name + " [" + displayName + "]";
}

File: sipXopenfire/config-plugin/src/org/sipfoundry/sipxconfig/web/rest/ChatRoom.java
package org.sipfoundry.sipxconfig.web.rest;

import java.util.List;

public class ChatRoom {
    private long id;
    private String name;
    private String description;
    private String subject;
    private boolean canChangeSubject;
    private String password;
    private boolean publicRoom;
    private boolean moderated;
    private boolean membersOnly;
    private int maxUsers;
    private boolean canInvite;
    private boolean canDiscoverJid;
    private boolean logEnabled;
    private int rolesToBroadcast;
    private List< ? > owners;
    private List< ? > members;

    public long getId() {
        return id;
    }

    public void setId(long id) {
        this.id = id;
    }

    public String getName() {
        return name;
    }

    public void setName(String name) {
        this.name = name;
    }

    public String getDescription() {
        return description;
    }

    public void setDescription(String description) {
        this.description = description;
    }

    public String getSubject() {
        return subject;
    }

    public void setSubject(String subject) {
        this.subject = subject;
    }

    public boolean isCanChangeSubject() {
        return canChangeSubject;
    }

    public void setCanChangeSubject(boolean canChangeSubject) {
        this.canChangeSubject = canChangeSubject;
    }

    public String getPassword() {
        return password;
    }

    public void setPassword(String password) {
        this.password = password;
    }

    public boolean isPublicRoom() {
        return publicRoom;
    }

    public void setPublicRoom(boolean publicRoom) {
        this.publicRoom = publicRoom;
    }

    public boolean isModerated() {
        return moderated;
    }

    public void setModerated(boolean moderated) {
        this.moderated = moderated;
    }

    public boolean isMembersOnly() {
        return membersOnly;
    }

    public void setMembersOnly(boolean membersOnly) {
        this.membersOnly = membersOnly;
    }

    public int getMaxUsers() {
        return maxUsers;
    }

    public void setMaxUsers(int maxUsers) {
        this.maxUsers = maxUsers;
    }

    public boolean isCanInvite() {
        return canInvite;
    }

    public void setCanInvite(boolean canInvite) {
        this.canInvite = canInvite;
    }

    public boolean isCanDiscoverJid() {
        return canDiscoverJid;
    }

    public void setCanDiscoverJid(boolean canDiscoverJid) {
        this.canDiscoverJid = canDiscoverJid;
    }

    public boolean isLogEnabled() {
        return logEnabled;
    }

    public void setLogEnabled(boolean logEnabled) {
        this.logEnabled = logEnabled;
    }

    public int getRolesToBroadcast() {
        return rolesToBroadcast;
    }

    public void setRolesToBroadcast(int rolesToBroadcast) {
        this.rolesToBroadcast = rolesToBroadcast;
    }

    public List< ? > getOwners() {
        return owners;
    }

    public void setOwners(List< ? > owners) {
        this.owners = owners;
    }

    public List< ? > getMembers() {
        return members;
    }

    public void setMembers(List< ? > members) {
        this.members = members;
    }

    @Override
    public String toString() {
        return "ChatRoom [id=" + id + ", name=" + name + ", description=" + description + ", subject=" + subject
            + ", canChangeSubject=" + canChangeSubject + ", password=" + password + ", publicRoom=" + publicRoom
            + ", moderated=" + moderated + ", membersOnly=" + membersOnly + ", maxUsers=" + maxUsers
            + ", canInvite=" + canInvite + ", canDiscoverJid=" + canDiscoverJid + ", logEnabled=" + logEnabled
            + ", rolesToBroadcast=" + rolesToBroadcast + ", owners=" + owners + ", members=" + members + "]";
    }
}
